bitmind/synthetic_image_generation/synthetic_image_generator.py

```diff
@@ -195,3 +203,83 @@ def generate_random_prompt(self, retry_attempts: int = 10) -> str:
             prompt += ', ' + np.random.choice(device, 1)[0]
             if prompt != "":
                 return prompt
+
+    def get_tokenizer_with_min_len(self):
+        """
+        Returns the tokenizer with the smallest maximum token length from the 'diffuser` object.
+    
+        If a second tokenizer exists, it compares both and returns the one with the smaller 
+        maximum token length. Otherwise, it returns the available tokenizer.
+        
+        Returns:
+            tuple: A tuple containing the tokenizer and its maximum token length.
+        """
+        # Check if a second tokenizer is available in the diffuser
+        if self.diffuser.tokenizer_2:
+            if self.diffuser.tokenizer.model_max_length > self.diffuser.tokenizer_2.model_max_length:
+                return self.diffuser.tokenizer_2, self.diffuser.tokenizer_2.model_max_length
+        return self.diffuser.tokenizer, self.diffuser.tokenizer.model_max_length
+
+    def truncate_prompt_if_too_long(self, prompt: str):
+        """
+        Truncates the input string if it exceeds the maximum token length when tokenized.
+    
+        Args:
+            prompt (str): The text prompt that may need to be truncated.
+    
+        Returns:
+            str: The original prompt if within the token limit; otherwise, a truncated version of the prompt.
+        """
+        tokenizer, max_token_len = self.get_tokenizer_with_min_len()
+        tokens = tokenizer(prompt, verbose=False) # Suppress token max exceeded warnings
+        if len(tokens['input_ids']) < max_token_len:
+            return prompt
+        # Truncate tokens if they exceed the maximum token length, decode the tokens back to a string
+        truncated_prompt = tokenizer.decode(token_ids=tokens['input_ids'][:max_token_len-1],
+                                            skip_special_tokens=True)
+        tokens = tokenizer(truncated_prompt)
+        bt.logging.info("Truncated prompt to abide by token limit.")
+        return truncated_prompt
+    
+    def generate_image(self, prompt, name = None, generate_at_target_size = False) -> list:
```

Comment:
Helper function to generate images. Useful for external calls where you want to use a prompt to generate one image.